File: EmployeeSystem/src/main/java/com/Backend/EmployeeSystem/service/EmpService.java
package com.Backend.EmployeeSystem.service;

import com.Backend.EmployeeSystem.entity.Emp;
import com.Backend.EmployeeSystem.payload.PostResponse;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.awt.print.Pageable;
import java.io.IOException;
import java.util.List;

public interface EmpService {

    public Emp createEmp(Emp emp,MultipartFile file) throws IOException;

    // Pagable
//    Page<Emp> getAllEmp(int page,int size);

    PostResponse getAllEmp(Integer pageNumber, Integer pageSize);

//   List<Emp> getAllEmp();

    public Emp getEmpById(Long id);
    public List<Emp> getEmpByFirstName(String keyword);
    public void deleteEmp(Long id);
    public void updateEmp(Long id, Emp emp, MultipartFile file) throws IOException;

}

File: EMS_Frontend/src/service/emp.service.js
import axios from "axios";

const BASE_API_URL = "http://localhost:8081/api/v1";

class EmpService {

    saveEmp(formdata) {
       return axios.post(BASE_API_URL + "/save", formdata)
           
    }

    getAllEmp(pageNumber, pageSize) {
        return axios.get(`${BASE_API_URL}/getAll`, {
            params: {
                pageNumber: pageNumber,
                pageSize: pageSize,
            },
        });
    }

    getEmpById(id) {
        return axios.get(BASE_API_URL + "/get/" + id);
    }

    deleteEmp(id) {
        return axios.delete(BASE_API_URL + "/delete/" + id);
    }

    updateEmp(id, formdata) {
        return axios.put(BASE_API_URL + "/update/" + id, formdata);
    }

    getEmpByFirstName(keyword){
        return axios.get(BASE_API_URL + "/gets/"+ keyword);
    }
}

export default new EmpService();